### PR TITLE
fix:[CDS-81957]: harness approval email notification line break fix

### PR DESCRIPTION
**Pull request by naman-goenka**
## Describe your changes
- adding a utility to convert new line characters to html compatible line break character. 
- We have two text fields (which can have new line characters) in approval summary: approval message and action (string generated when approval is approved or rejected).
- only for email notification in harness approvals, this is being done.

Testing: 
a. email when approval start with execution details
Screenshot 2023-10-31 at 10 53 26 PM

b. email when approval start without execution details
Screenshot 2023-10-31 at 10 58 39 PM

c. email when approver approves without execution details
Screenshot 2023-10-31 at 10 59 59 PM

d. email when approver approves with execution details
Screenshot 2023-10-31 at 10 55 35 PM

e. email when approver rejects without execution details
Screenshot 2023-10-31 at 11 01 01 PM

f. email when approver rejects with execution details

Screenshot 2023-10-31 at 10 56 42 PM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54865)

### CODE REVIEW

**Comment by naman-goenka** [comment]
trigger smartchecks

**Comment by naman-goenka** [comment]
trigger uts

**Comment by naman-goenka** [comment]
trigger uts

**Comment by naman-goenka** [comment]
trigger smartchecks

**Comment by naman-goenka** [comment]
trigger pmd

File: pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java [hunk 1]
```diff
@@ -401,11 +406,11 @@ private String getAction(HarnessApprovalInstance approvalInstance) {
         for (HarnessApprovalActivity harnessApprovalActivity : harnessApprovalActivities) {
           String userIdentification = getUserIdentification(harnessApprovalActivity.getUser());
           action = action
-              + (userIdentification + " approved on " + formatTime(harnessApprovalActivity.getApprovedAt()) + "   \\n");
+              + (userIdentification + " approved on " + formatTime(harnessApprovalActivity.getApprovedAt()) + "   \n");
         }
         if (!isEmpty(action)) {
           // removing last redundant new line character
-          action = action.substring(0, action.length() - 2);
+          action = action.substring(0, action.length() - 1);
```
Comment (sribalij) [inline pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java:413]: Does this still have to be -1 or -2?

Comment (sribalij) [inline pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java:413]: Please include Junit for this test case as well

Comment (naman-goenka) [reply pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java:413]: This case is covered in the assertion made for `action` in modified test case

Comment (naman-goenka) [reply pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java:413]: This should be `-1` to just remove the last newline character. Earlier `-2` was being done for removing `\` and `\n` .

Comment (naman-goenka) [reply pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java:413]: For slack also rendering seems fine for action notification with `\n` 
Screenshot 2023-11-01 at 11 34 53 AM

**Comment by naman-goenka** [comment]
trigger gitleaks

**Review by abhinavmittal-harness — DISMISSED** [review]
> Reviewed 3 of 5 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54865)* status: 5 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/ApprovalUtils.java` line 200 at r2](https://reviewable.io/reviews/harness/harness-core/54865#-NiAOYF8AwvKudpDvHGw:-NiAOYF93PWW9XdIJ9x6:b-7ua8jo) ([raw file](https://github.com/harness/harness-core/blob/858643e4b91437fb06ff5770adbe82985f7d3b62/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/ApprovalUtils.java#L200)):*
> ```Java
>               ->
>           // regex: non-capturing group with 3 alternatives matching carriage return and newline characters
>           value.replaceAll("(?:\\r\\n|\\r|\\n)", "<br>"));
> ```

nit: we can make this constant

**Comment by naman-goenka** [comment]
trigger uts

**Comment by naman-goenka** [comment]
> > Reviewed 3 of 5 files at r1, 2 of 2 files at r2, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/54865)_ status: 5 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.
> 
> _[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/ApprovalUtils.java` line 200 at r2](https://reviewable.io/reviews/harness/harness-core/54865#-NiAOYF8AwvKudpDvHGw:-NiAOYF93PWW9XdIJ9x6:b-7ua8jo) ([raw file](https://github.com/harness/harness-core/blob/858643e4b91437fb06ff5770adbe82985f7d3b62/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/ApprovalUtils.java#L200)):_
> 
> > ```java
> >               ->
> >           // regex: non-capturing group with 3 alternatives matching carriage return and newline characters
> >           value.replaceAll("(?:\\r\\n|\\r|\\n)", "<br>"));
> > ```
> 
> nit: we can make this constant

done

**Review by sourabh-awashti — APPROVED** [review]
LGTM

**Review by abhinavmittal-harness — APPROVED** [review]
LGTM

**Comment by naman-goenka** [comment]
trigger review

**Pull request merged**